### ISSUE TITLE
esp_http_client_perform is failing when called with the same config in a new stack frame

### ISSUE THREAD

**Issue by jdbaptista** [zinkwazi/traffic-pcb-sb#13]
Not sure why this is happening yet, but two requests to either the same or different LEDs fails.

**Comment by jdbaptista** [comment]
This is caused by stack smashing in the http event handler with the messy logic I wrote and didn't effectively test, whoops. Moral of this story is to test code and turn on stack canaries while testing.
```
I (2354) esp-x509-crt-bundle: Certificate validated
E (3594) TomTom: tomtomRequestPerform(464): recieved an error from the http client event handler
E (3594) TomTom: tomtomRequestSpeed(182): failed to perform API request

Stack smashing protect failure!


Backtrace: 0x4008182e:0x3ffba630 0x40089719:0x3ffba650 0x400811de:0x3ffba670 0x400d7865:0x3ffba690 0x400d719d:0x3ffba770 0x401705b3:0x3ffba880 0x4008a2d1:0x3ffba8b0
```

**Comment by jdbaptista** [comment]
Stack smashing was due to an off by one error in URL_LENGTH, which is now fixed as of cf6884a.
